src/js/integrations/reactnativeerrorhandlers.ts

```diff
@@ -58,15 +58,28 @@ export class ReactNativeErrorHandlers implements Integration {
     if (this._options.onunhandledrejection) {
       /*
         In newer RN versions >=0.63, the global promise is not the same reference as the one imported from the promise library.
-        This is due to a version mismatch between promise versions. The version will need to be fixed with a package resolution.
-        We first run a check and show a warning if needed.
+        This is due to a version mismatch between promise versions.
       */
-      this._checkPromiseVersion();
+      this._polyfillPromise();
+    }
+  }
+  /**
+   * Polyfill the global promise instance with one we can be sure that we can attach the tracking to.
```

Comment:
Would love some code comments that essentially explain how we got here. The different promise patching strats we used, and why is is the best one.

will help the forensics in the future lol 🕵️‍♂️